internal/ethapi/api.go

```diff
@@ -1323,7 +1323,7 @@ func newRPCTransaction(tx *types.Transaction, blockHash common.Hash, blockNumber
 			price := math.BigMin(new(big.Int).Add(tx.GasTipCap(), baseFee), tx.GasFeeCap())
 			result.GasPrice = (*hexutil.Big)(price)
 		} else {
-			result.GasPrice = nil
+			result.GasPrice = (*hexutil.Big)(tx.GasFeeCap())
```

Comment:
Actually, if you don't set it to anything, it will be the fee cap as that's returned by `GasPrice` that's set above for all txs.

Reply:
@karalabe do you prefer removing this line and rely on the initial definition of `RPCTransaction` above or defining the result explicitly?

Reply:
Guess it's fine like this too